internal/ast/compiler/disjunctions.go

```diff
@@ -197,14 +194,6 @@ func (pass *DisjunctionToType) processStruct(schema *ast.Schema, def ast.Type) (
 func (pass *DisjunctionToType) processDisjunction(schema *ast.Schema, def ast.Type) (ast.Type, error) {
 	disjunction := def.AsDisjunction()
 
-	// Ex: type | null
-	if len(disjunction.Branches) == 2 && disjunction.Branches.HasNullType() {
-		finalType := disjunction.Branches.NonNullTypes()[0]
-		finalType.Nullable = true
-
-		return finalType, nil
-	}
```

Comment:
This behavior was interesting to generate cleaner types in Python, so I moved it out into its own compiler pass: [internal/ast/compiler/disjunctions_with_null_to_optional.go](https://github.com/grafana/cog/pull/195/files#diff-30adea340ca2f95cba1a878f783a7eef6d72a2d1aa748c43132785c54f0044fd)